CONTRIBUTING.md

```diff
@@ -113,6 +113,46 @@ the repository maintainers will integrate it, prepare changelogs, and
 ensure it gets released in one of upcoming Weekly releases.
 There is no additional action required from pull request authors at this point.
 
+### Pull requests management
+
+The project is using some labels to mark the status and the content of the pull requests.
+The complete list of labels can be found at https://github.com/jenkinsci/jenkins/labels.
+Out of those labels, here is a definition of labels the maintainers are using to manage the status of the pull requests:
+
+- `needs-docs` marks a pull requests as lacking documentation, either for developers (e.g., Javadoc) or users (e.g., changes the [Jenkins handbook](https://www.jenkins.io/doc/book/)).
+For such pull requests to be approved and merged, the corresponding changes to the documentation should be proposed.
+If those changes belong to a separate repository (e.g., `jenkins-infra/jenkins.io`), a secondary pull request should be created in draft state in the other repository and review in tandem with the primary pull request that proposes the code change.
+- `needs-fix` marks a pull requests with reviews requesting some code change which are not addressed yet.
+Such pull requests won't be merged until the code has been fixed and the tests are passing
+- `needs-justification` marks pull requests on which maintainers are debating the motivation for the proposed changed.
+The maintainers are requiring a clear description of the high-level effort with which the pull request is associated.
+This is to ensure that the context of the changes is well understood by everyone.
+The use of design document, high-level tracking epics, minimal reproducible example with steps, etc. is encouraged.
+- `needs-more-review` marks pull requests which are lacking reviews and comments, because the changes are complex or because a debate started among reviewers and more opinions would be beneficial
```

Comment:
```suggestion
`needs-more-review`:: The pull request lacks a sufficient number of review from subject-matter expert(s) (SME), either because the changes are complex and not sufficiently explained or because there is a lack of consensus regarding the proposed solution.
```